modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/ImplicitConfigurationProviderFactory.java

```diff
@@ -39,6 +37,17 @@ ConfigurationProvider createImplicitConfigurationProvider(ExtensionModel extensi
                                                             ConfigurationModel implicitConfigurationModel,
                                                             CoreEvent muleEvent,
                                                             ReflectionCache reflectionCache,
-                                                            ExpressionManager expressionManager,
-                                                            MuleContext muleContext);
+                                                            ExpressionManager expressionManager);
+
+  /**
+   * Returns an implicit {@link ConfigurationProvider} name, that can be used to look it up.
+   *
+   * @param extensionModel     The configurable {@link ExtensionModel}.
+   * @param configurationModel The {@link ConfigurationModel} that represents the extensionModel configuration.
+   * @return The {@link ConfigurationProvider} name.
+   * @see org.mule.runtime.module.extension.internal.manager.ExtensionRegistry#getConfigurationProvider(String)
+   * @since 4.4.0
+   */
+  String resolveImplicitConfigurationProviderName(ExtensionModel extensionModel, ConfigurationModel configurationModel);
```

Comment:
why is this moved into this factory and exposed in this interface? it's weird that we have the need to expose this as an interface method. Sounds like an implementation detail that shouldn't be exposed by a factory class. This is a code smell. Double check if this responsibility doesn't belong somewhere else. I don't understand the reason why it should be moved here